changes/2023-09-26_encryption-context-decrypt/background.md

```diff
@@ -0,0 +1,540 @@
+var reqCMM :- expect mpl.CreateRequiredEncryptionContextCMM(
+   mplTypes.CreateRequiredEncryptionContextCMMInput(
+         underlyingCMM := Some(defaultCMM),
+         keyring := None,
+         requiredEncryptionContextKeys := requiredEncryptionContextKeys
+   )
+);
+
+var decryptOutput := esdk.Decrypt(Types.DecryptInput(
+   ciphertext := esdkCiphertext,
+   materialsManager := Some(reqCMM),
+   keyring := None,
+   encryptionContext := Some(reproducedEncryptionContext)
+));
+
+expect decryptOutput.Success?;
+var cycledPlaintext := decryptOutput.value.plaintext;
+expect cycledPlaintext == asdf;
+```
+
+### Test Different Encryption Context on Decrypt Failure
+
+Encrypt with and store all encryption context in the message header.
+On Decrypt will supply additional encryption context not stored in the header.
+This operation MUST fail.
+
+```
+// The string "asdf" as bytes
+var asdf := [ 97, 115, 100, 102 ];
+var aesKeyring := GetAesKeyring();
+
+// Test supply same encryption context on encrypt and decrypt NO filtering
+var encryptionContext := map[ keyA := valA, keyB := valB ];
+var reproducedAdditionalEncryptionContext := map[ keyC := valC ];
+var requiredEncryptionContextKeys := [keyA, keyB];
+
+var defaultCMM :- expect mpl.CreateDefaultCryptographicMaterialsManager(
+   mplTypes.CreateDefaultCryptographicMaterialsManagerInput(
+         keyring := aesKeyring
+   )
+);
+
+var encryptOutput := esdk.Encrypt(Types.EncryptInput(
+   plaintext := asdf,
+   encryptionContext := Some(encryptionContext),
+   materialsManager := Some(defaultCMM),
+   keyring := None,
+   algorithmSuiteId := None,
+   frameLength := None
+));
+
+expect encryptOutput.Success?;
+var esdkCiphertext := encryptOutput.value.ciphertext;
+
+// Create Required EC CMM with the required EC Keys we want
+var reqCMM :- expect mpl.CreateRequiredEncryptionContextCMM(
+   mplTypes.CreateRequiredEncryptionContextCMMInput(
+         underlyingCMM := Some(defaultCMM),
+         keyring := None,
+         requiredEncryptionContextKeys := requiredEncryptionContextKeys
+   )
+);
+
+var decryptOutput := esdk.Decrypt(Types.DecryptInput(
+   ciphertext := esdkCiphertext,
+   materialsManager := Some(reqCMM),
+   keyring := None,
+   encryptionContext := Some(reproducedAdditionalEncryptionContext)
+));
+
+expect decryptOutput.Failure?;
+```
+
+### Test MisMatched Encryption Context on Decrypt Failure
+
+Encrypt with and store all encryption context in the message header.
+On Decrypt will supply mismatched encryption context that is stored in the header.
+This operation MUST fail.
+
+```
+// The string "asdf" as bytes
+var asdf := [ 97, 115, 100, 102 ];
+var aesKeyring := GetAesKeyring();
+
+// Test supply same encryption context on encrypt and decrypt NO filtering
+var encryptionContext := map[ keyA := valA, keyB := valB ];
+var reproducedAdditionalEncryptionContext := map[ keyA := valB, keyB := valA  ];
+var requiredEncryptionContextKeys := [keyA, keyB];
```

Comment:
I thought this was testing context that _is_ in the header?

Reply:
yup good catch